.github/workflows/release.yml

```diff
@@ -12,22 +12,32 @@ permissions:
 
 jobs:
   release:
+    if: "startsWith(github.ref, 'refs/tags/v')"
     runs-on: ubuntu-latest
-
     steps:
       - name: Checkout code
         uses: actions/checkout@v4
         with:
           fetch-depth: 0
 
+      - name: Add Version as environment veriable
+        run: |
+          VERSION=$(echo "${GITHUB_REF#refs/tags/v}")
+          echo "VERSION=${VERSION}" >> $GITHUB_ENV
+    
+      - name: Add version to manifest file
+        run: |
+          # Change $VERSION in the manifest.yml to the version of the release
+          envsubst < manifest.yml > manifest_temp.yml && mv manifest_temp.yml manifest.yml
+
       - name: Set up Go
         uses: actions/setup-go@v5
```

Comment:
Nit: specify a Go version instead of using the default to ensure consistency across different runners.
```
- name: Set up Go
  uses: actions/setup-go@v5
  with:
    go-version: '1.22'  # Specify your required Go version
```